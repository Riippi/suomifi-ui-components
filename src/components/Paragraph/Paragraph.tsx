import React, { Component } from 'react';
import classnames from 'classnames';
import { HtmlP, HtmlPProps } from '../../reset/HtmlP/HtmlP';

export interface ParagraphProps extends HtmlPProps {
  className?: string;
}

export const baseClassName = 'fi-paragraph';

export class Paragraph extends Component<ParagraphProps> {
  render() {
    const { className, ...passProps } = this.props;
    return (
      <HtmlP {...passProps} className={classnames(baseClassName, className)} />
    );
  }
}
