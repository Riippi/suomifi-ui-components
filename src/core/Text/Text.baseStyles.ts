import { css } from 'styled-components';
import { suomifiTheme } from '../theme';
import { TextProps } from './Text';
import { element, fonts } from '../theme/reset';
import { objValue } from '../../utils/typescript';

export const baseStyles = ({ theme = suomifiTheme, color }: TextProps) => css`
  ${element(theme)}
  ${fonts(theme).body}
  color: ${!!color ? objValue(theme.colors, color) : theme.colors.blackBase};

  &.fi-text {
    &--bold {
      ${fonts(theme).semiBold}
    }
    &--lead {
      ${fonts(theme).lead}
    }
    &--small-screen {
      ${fonts(theme).smRes.body}
      &.fi-text--bold {
        ${fonts(theme).smRes.semiBold}
      }
      &.fi-text--lead {
        ${fonts(theme).smRes.lead}
      }
    }
  } 
`;
