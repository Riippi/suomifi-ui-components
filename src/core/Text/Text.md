```js
import { Paragraph, Text } from 'suomifi-ui-components';

<>
  <Paragraph>Paragraph text</Paragraph>
  <Paragraph marginBottomSpacing="m">
    <Text.lead>Leading text</Text.lead>
  </Paragraph>
  <Paragraph>
    <Text>Body text</Text> <Text.bold>Bold text</Text.bold>
  </Paragraph>
  <Paragraph>
    <Text smallScreen={true}>Body text</Text>{' '}
    <Text.bold smallScreen={true}>Bold text</Text.bold>
  </Paragraph>
</>;
```
