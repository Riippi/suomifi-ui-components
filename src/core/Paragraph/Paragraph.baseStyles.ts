import { css } from 'styled-components';
import { suomifiTheme } from '../theme';
import { ParagraphProps } from './Paragraph';
import { element, fonts } from '../theme/reset';
import { objValue } from '../../utils/typescript';
import { margin } from '../theme/utils/spacing';

export const baseStyles = ({
  theme = suomifiTheme,
  color,
  marginBottomSpacing = '0',
}: ParagraphProps) => css`
  ${element(theme)}
  ${fonts(theme).body}
  ${margin(theme)('0', '0', marginBottomSpacing, '0')};
  color: ${!!color ? objValue(theme.colors, color) : theme.colors.blackBase};
`;
