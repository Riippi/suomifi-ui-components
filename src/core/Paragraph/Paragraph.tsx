import React, { Component } from 'react';
import { default as styled } from 'styled-components';
import { withDefaultTheme } from '../theme/utils';
import { ThemeComponent, ColorProp } from '../theme';
import {
  Paragraph as CompParagraph,
  ParagraphProps as CompParagraphProps,
} from '../../components/Paragraph/Paragraph';
import { baseStyles } from './Paragraph.baseStyles';
import { SpaceProp } from '../theme/utils/spacing';

export interface ParagraphProps extends CompParagraphProps, ThemeComponent {
  /** Change color */
  color?: ColorProp;
  /** Spacing token for bottom margin */
  marginBottomSpacing?: SpaceProp;
}

const StyledParagraph = styled(
  ({ theme, color, marginBottomSpacing, ...passProps }: ParagraphProps) => (
    <CompParagraph {...passProps} />
  ),
)`
  ${props => baseStyles(props)};
`;

/**
 * Used displaying Paragraph with correct styles
 */
export class Paragraph extends Component<ParagraphProps> {
  render() {
    return <StyledParagraph {...withDefaultTheme(this.props)} />;
  }
}
