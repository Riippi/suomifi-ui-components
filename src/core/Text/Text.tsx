import React, { Component } from 'react';
import { default as styled } from 'styled-components';
import { withDefaultTheme } from '../theme/utils';
import { ThemeComponent, ColorProp } from '../theme';
import {
  Text as CompText,
  TextProps as CompTextProps,
} from '../../components/Text/Text';
import { baseStyles } from './Text.baseStyles';
import classnames from 'classnames';

const baseClassName = 'fi-text';
const smallScreenClassName = `${baseClassName}--small-screen`;

export interface TextProps extends CompTextProps, ThemeComponent {
  /** Change font to smaller screen size and style */
  smallScreen?: boolean;
  /** Change color for text from theme colors */
  color?: ColorProp;
  /**
   * Type of text-style
   * @default body
   */
  variant?: 'body' | 'lead' | 'bold';
}

const StyledText = styled(
  ({
    theme,
    color,
    variant = 'body',
    smallScreen,
    className,
    ...passProps
  }: TextProps) => (
    <CompText
      {...passProps}
      className={classnames(className, [`${baseClassName}--${variant}`], {
        [smallScreenClassName]: smallScreen,
      })}
    />
  ),
)`
  ${props => baseStyles(props)};
`;

/**
 * Used displaying text with correct fonts
 */
export class Text extends Component<TextProps> {
  static lead = (props: TextProps) => (
    <StyledText {...withDefaultTheme(props)} variant="lead" />
  );

  static bold = (props: TextProps) => (
    <StyledText {...withDefaultTheme(props)} variant="bold" />
  );

  render() {
    const passProps = withDefaultTheme(this.props);
    return <StyledText {...passProps} />;
  }
}
