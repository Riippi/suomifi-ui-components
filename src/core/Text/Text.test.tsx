import React from 'react';
import { render } from 'react-testing-library';
import { axeTest } from '../../utils/test/axe';

import { Paragraph } from '../Paragraph/Paragraph';
import { Text } from './Text';

const TestTexts = (
  <div data-testid="test-text">
    <Paragraph>Paragraph text</Paragraph>
    <Paragraph marginBottomSpacing="m">
      <Text.lead>Leading text</Text.lead>
    </Paragraph>
    <Paragraph>
      <Text>Body text</Text> <Text.bold>Bold text</Text.bold>
    </Paragraph>
    <Paragraph>
      <Text smallScreen={true}>Body text</Text>{' '}
      <Text.bold smallScreen={true}>Bold text</Text.bold>
    </Paragraph>
  </div>
);

test('calling render with the same component on the same container does not remount', () => {
  const TextRendered = render(TestTexts);
  const { container } = TextRendered;
  expect(container.firstChild).toMatchSnapshot();
});

test('should not have basic accessibility issues', axeTest(TestTexts));
