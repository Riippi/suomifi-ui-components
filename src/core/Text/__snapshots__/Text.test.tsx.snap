// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calling render with the same component on the same container does not remount 1`] = `
.c1 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0 {
  color: hsl(0,0%,16%);
  font-family: 'Source Sans Pro','Helvetica Neue',Arial,sans-serif;
  font-size: 18px;
  font-weight: 400;
  line-height: 1.5;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  color: hsl(0,0%,16%);
}

.c0.fi-text--bold {
  font-family: 'Source Sans Pro','Helvetica Neue',Arial,sans-serif;
  font-size: 18px;
  font-weight: 600;
  line-height: 1.5;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
}

.c0.fi-text--lead {
  font-family: 'Source Sans Pro','Helvetica Neue',Arial,sans-serif;
  font-size: 22px;
  font-weight: 400;
  line-height: 1.5;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
}

.c0.fi-text--small-screen {
  font-family: 'Source Sans Pro','Helvetica Neue',Arial,sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
}

.c0.fi-text--small-screen.fi-text--bold {
  font-family: 'Source Sans Pro','Helvetica Neue',Arial,sans-serif;
  font-size: 16px;
  font-weight: 600;
  line-height: 1.5;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
}

.c0.fi-text--small-screen.fi-text--lead {
  font-family: 'Source Sans Pro','Helvetica Neue',Arial,sans-serif;
  font-size: 20px;
  font-weight: 400;
  line-height: 1.5;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
}

<div
  data-testid="test-text"
>
  <span
    class="fi-text c0 fi-text--body c1"
  >
    Hey this is test
  </span>
  <span
    class="fi-text c0 fi-text--lead c1"
  >
    Hey this is test lead
  </span>
  <span
    class="fi-text c0 fi-text--bold c1"
  >
    Hey this is test bold
  </span>
</div>
`;
